### PR TITLE
feat: make agent stats' cardinality configurable

### PR DESCRIPTION
**Pull request by dannykopping**
Part 2 of https://github.com/coder/coder/pull/12468

We initially had to [revert](https://github.com/coder/coder/pull/12533) the above PR because of a panic from Prometheus.

This was caused by the 4 default labels being passed in to `AgentStats()`, however these metrics do not contain a value for the `template_name` label. This was masked in a test by passing in `nil` for the aggregation labels which defaulted them to the 3 correct ones, but I subsequently changed the default value from `nil` in the actual code to all 4 labels for various reasons.

I implemented the fix in 3c1d016a82d640dbcb258fc3576899b7678a928e. I can't think of a cleaner approach than this; if you come up with a better approach, please share!

**Note to reviewers:** the code is otherwise identical to #12468, so simply reviewing 3c1d016a82d640dbcb258fc3576899b7678a928e should suffice.

### CODE REVIEW

File: coderd/prometheusmetrics/aggregator.go [hunk 1]
```diff
@@ -8,8 +8,11 @@ import (
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
 	"golang.org/x/xerrors"
 
+	"github.com/coder/coder/v2/coderd/agentmetrics"
```
Comment (johnstcn) [inline coderd/prometheusmetrics/aggregator.go:14]: nit: sorting

**Review by johnstcn — APPROVED** [review]
Only a couple of nits from me. Tested locally with a small scaletest.

File: coderd/prometheusmetrics/prometheusmetrics_test.go [hunk 2]
```diff
@@ -11,6 +11,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/coder/coder/v2/coderd/agentmetrics"
+
 	"github.com/coder/coder/v2/coderd/batchstats"
 	"github.com/coder/coder/v2/coderd/database/dbtestutil"
 	"github.com/coder/coder/v2/coderd/database/dbtime"
```
[the remark this answers is not in the document]
Comment (dannykopping) [reply coderd/prometheusmetrics/prometheusmetrics_test.go:18]: Hhmm, I ran `make fmt` and it didn't correct it so I think we have some things to fix in our CI. Will manually sort though :+1:

[the remark this answers is not in the document]
Comment (dannykopping) [reply coderd/prometheusmetrics/prometheusmetrics_test.go:18]: Oh nice. I just ran it and it... changes _a lot_.
I'll raise a separate PR.

[the remark this answers is not in the document]
Comment (johnstcn) [reply coderd/prometheusmetrics/prometheusmetrics_test.go:18]: It is _extremely_ opinionated

**Review by mtojek — APPROVED** [review]
👍

**Pull request merged**